### ISSUE TITLE
Question regarding the Implementation of Integrated Gradients 

### ISSUE THREAD

**Issue by champnaman** [albermax/innvestigate#156]
Hey Alber,

I was reading the Integrated Gradients (IG) paper and my understanding is that they backpropagate through a model **with** softmax layer. I also looked at their code to verify this and the [same was mentioned there](https://github.com/ankurtaly/Integrated-Gradients/blob/master/howto.md#identifying-the-output-tensor) as well. 
![image](https://user-images.githubusercontent.com/5251592/61961908-03792a80-af8e-11e9-9585-7fa86d9253f7.png)

But looking at your implementation, my understanding is that you pass a `model_wo_softmax` for analyzing. It would be great if you could help me understand as to why this discrepancy is happening or did I miss something?

Thanks, 
Naman

**Comment by enryH** [comment]
It is briefly mentioned in the appendix under A, if I see it correctly. Reference given is:

Montavon, G., Lapuschkin, S., Binder, A., Samek, W., Müller, K.R.: Explaining nonlinear classification decisions with deep Taylor decomposition. Pattern Recogn. 65, 211–222 (2017)

**Comment by sebastian-lapuschkin-sideprojects** [comment]
Hi @champnaman:

TL;DR: LRP-z = LRP-eps with eps=0

**Comment by sebastian-lapuschkin** [comment]
Hi @champnaman:

TL;DR: LRP-z = LRP-eps with eps=0

**Comment by sebastian-lapuschkin** [comment]
Hi @champnaman:

**TL;DR:**
LRP-z = LRP-eps with eps=0.
LRP-z can be regarded as the most basic (naive) decomposition approach

**Comment by champnaman** [comment]
Hi @sebastian-lapuschkin 

Thank you and @enryH for your responses regarding my second question. 
Could you please help me clear the confusion in the first question as well?

Thanks, 
Naman

**Comment by sebastian-lapuschkin** [comment]
There is a group of methods, which avoids using the softmax for explanation, e.g. LRP, Sensitivity Analysis, Deconv (afaik) and some others.

There is an answer/explanation [here](https://github.com/albermax/innvestigate/issues/84) why those methods remove the softmax.

Hope this answers the remaining question ;)

**Comment by sebastian-lapuschkin** [comment]
plus, in addition to above link:

softmax(x) = softmax(x+c) with c in R
